catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -240,27 +347,36 @@ private void writeResourceToZip(
 
   private BinaryContent getBinaryContentFromZip(String filePath)
       throws CatalogTransformerException {
-    BinaryContent binaryContent;
-    InputStream fileInputStream;
+    byte[] bytes;
     File zipFile = new File(filePath);
     try {
-      fileInputStream = new ZipInputStream(new FileInputStream(zipFile));
-      binaryContent = new BinaryContentImpl(fileInputStream);
-    } catch (FileNotFoundException e) {
-      throw new CatalogTransformerException("Unable to get ZIP file from ZipInputStream.", e);
+      bytes = Files.readAllBytes(zipFile.toPath().toAbsolutePath());
```

Comment:
❗️ I'm working on a [kind of similar PR](https://github.com/codice/ddf/pull/4220), where I am trying to encrypt/decrypt data we read from/write to disk. @pklinef brought up a good point about passing byte arrays around: the resources we are dealing with may be too large to fit into memory.

His recommendation was to implement everything using streams. However, I would get confirmation from Phil since I'm not 100% sure it applies here.

Reply:
It looks like this could be cause an out of memory exception.  That could be used in a denial of service attack.  See the discussion on the other PR that Jacob linked to or see this similar ticket.

https://codice.atlassian.net/browse/DDF-937

@codice/security